pkg/peer/state/peer_state.go

```diff
@@ -0,0 +1,72 @@
+package state
+
+import (
+	"sync"
+
+	"github.com/matrix-org/waterfall/pkg/common"
+	"github.com/pion/webrtc/v3"
+)
+
+type RemoteTrackId struct {
+	id        string
+	simulcast common.SimulcastLayer
+}
+
+type PeerState struct {
+	mutex        sync.Mutex
+	dataChannel  *webrtc.DataChannel
+	remoteTracks map[RemoteTrackId]*webrtc.TrackRemote
+}
+
+func NewPeerState() *PeerState {
+	return &PeerState{
+		remoteTracks: make(map[RemoteTrackId]*webrtc.TrackRemote),
+	}
+}
+
+func (p *PeerState) AddRemoteTrack(track *webrtc.TrackRemote) {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	p.remoteTracks[RemoteTrackId{track.ID(), common.RIDToSimulcastLayer(track.RID())}] = track
+}
+
+func (p *PeerState) RemoveRemoteTrack(track *webrtc.TrackRemote) {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	delete(p.remoteTracks, RemoteTrackId{track.ID(), common.RIDToSimulcastLayer(track.RID())})
+}
+
+func (p *PeerState) GetRemoteTrack(id string, simulcast common.SimulcastLayer) *webrtc.TrackRemote {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	return p.remoteTracks[RemoteTrackId{id, simulcast}]
+}
+
+func (p *PeerState) GetRemoteTracks() []*webrtc.TrackRemote {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	tracks := make([]*webrtc.TrackRemote, 0, len(p.remoteTracks))
+	for _, track := range p.remoteTracks {
+		tracks = append(tracks, track)
+	}
+
+	return tracks
+}
```

Comment:
Seems to be unused

Reply:
Yep, I just created it for completeness (since it's in a separate package), but I can remove unused functions. The idea of the `PeerState` was to isolate mutex-protected data in its own structure to prevent users from accessing the fields without locking the mutex. Essentially, we isolate it to ensure that we access things in a thread-safe way.

Reply:
Yeah, I think we should avoid unused methods...

Reply:
👌 Removed.